### ISSUE TITLE
Blob length off by one

### ISSUE THREAD

**Issue by donv** [SQLDroid/SQLDroid#24]
Running the blob tests has revealed that blobs report a length that is one too high.

**Comment by jhannes** [comment]
This seems to have been addressed in c6a59b700c81c223936f2d38aef13d42cf1f91ca. *But I think that rather than fixing the issue, this introduced a new bug, where the Blob is one byte too _short_*

I've added some tests to try to find the bug, but instead, there seems to be a new bug created by this commit. See https://github.com/SQLDroid/SQLDroid/blob/faf011d7a3329ea1274fd79ef12ba7f48bdcbd4b/src/test/java/org/sqldroid/SQLDroidTest2.java#L214, which fails with the current implementation of https://github.com/SQLDroid/SQLDroid/blob/faf011d7a3329ea1274fd79ef12ba7f48bdcbd4b/src/main/java/org/sqldroid/SQLDroidResultSet.java#L258

These tests are run under Robolectric, i.e. not on the actual device. Either this fix was bad or the behavior of Cursor.getBlob varies from Robolectric to native device.

**Comment by donv** [comment]
I think the fix was correct, at least for my use cases.  It is a while ago, so I cannot remember clearly, but I think we got an extra zero-byte compared to the string we actually stored.

Hopefully we can get the same test suite green in Robolectric and in an emulator to be sure.

**Comment by jhannes** [comment]
You're right. I was able to reproduce your result: Under Android, String.getBytes() returns a final \0-byte, at least in this case.
